federationapi/routing/send.go

```diff
@@ -275,6 +280,13 @@ func (t *txnReq) processTransaction(ctx context.Context) (*gomatrixserverlib.Res
 			continue
 		}
 
+		// Clear our local user profile cache, if this is a membership event
```

Comment:
It would actually be better for the user API to have a consumer for the output room events topic and follow that, rather than hooking into the federation API. The roomserver is the single source of truth for the member state.